src/visitor.rs

```diff
@@ -268,6 +268,11 @@ impl TypeRef {
     /// dictionary, it recursively calls itself on their underlying element, key, and value types.
     pub fn visit_with(&self, visitor: &mut impl Visitor) {
         visitor.visit_type_ref(self);
+
+        if matches!(&self.definition, TypeRefDefinition::Unpatched(_)) {
```

Comment:
Just beat me by a minute : vP
I also had this comment explaining it:
```suggestion
        // If this typeref isn't patched, do not attempt to visit it further.
        // Note that sequence and dictionary types (the only types we visit further) are always patched anyways.
        if matches!(&self.definition, TypeRefDefinition::Unpatched(_)) {
```